cmd/kubectl_token.go

```diff
@@ -175,76 +173,107 @@ func deleteCachedCredential(ctx *cli.Context) error {
 	if len(ctx.Args()) == 0 {
 		return cli.ShowSubcommandHelp(ctx)
 	}
-	dir, err := os.Getwd()
+
+	sc, err := lookupServerConfig(ctx)
 	if err != nil {
 		return err
 	}
-	cacheDir := filepath.Join(dir, kubeConfigCache)
+
+	// dir is always set by global default.
+	dir := ctx.GlobalString("config")
+
+	if len(sc.KubeCredentials) == 0 {
+		customPrint(fmt.Sprintf("there are no cached tokens in [%s]", dir))
+		return nil
+	}
+
 	if ctx.Args().First() == "all" {
-		customPrint(fmt.Sprintf("removing cached tokens [%s]", cacheDir))
-		return os.RemoveAll(cacheDir)
+		customPrint(fmt.Sprintf("removing cached tokens [%s]", dir))
+		sc.KubeCredentials = make(map[string]*config.ExecCredential)
```

Comment:
when does `sc` get written to config file?

Reply:
I missed this, many apologies.